Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -0,0 +1,480 @@
+                return;
+
+            auto options = adoptNS([PAL::allocWPResourceRequestOptionsInstance() init]);
+            [options setAfterUpdates:YES];
+
+            [[PAL::getWPResourcesClass() sharedInstance] requestTrackerNetworkAddresses:options.get() completionHandler:^(NSArray<WPNetworkAddressRange *> *ranges, NSError *error) {
+                if (error) {
+                    RELEASE_LOG_ERROR(ResourceLoadStatistics, "Failed to request tracking IP addresses from WebPrivacy");
+                    return;
+                }
+
+                version4List().clear();
+                version6List().clear();
+
+                for (WPNetworkAddressRange *range in ranges) {
+                    switch (range.version) {
+                    case WPNetworkAddressVersion4:
+                        version4List().append({ range });
+                        break;
+                    case WPNetworkAddressVersion6:
+                        version6List().append({ range });
+                        break;
+                    default:
+                        RELEASE_LOG_ERROR(ResourceLoadStatistics, "Skipped invalid tracking IP address: %@", range);
+                        break;
+                    }
+                }
+
+                version4List().shrinkToFit();
+                version6List().shrinkToFit();
+            }];
+        });
+    }
+
+    static const TrackerAddressLookupInfo* find(const WebCore::IPAddress& address)
+    {
+        auto& list = address.isIPv4() ? version4List() : version6List();
+        if (list.isEmpty())
+            return nullptr;
+
+        size_t lower = 0;
+        size_t upper = list.size() - 1;
+        if (address < list[lower].m_network)
+            upper = lower;
+        else if (address > list[upper].m_network)
+            lower = upper;
+        else {
+            while (upper - lower > 1) {
+                auto middle = (lower + upper) / 2;
+                switch (address.compare(list[middle].m_network)) {
+                case WebCore::IPAddress::ComparisonResult::Equal:
+                    return &list[middle];
+                case WebCore::IPAddress::ComparisonResult::Less:
+                    upper = middle;
+                    break;
+                case WebCore::IPAddress::ComparisonResult::Greater:
+                    lower = middle;
+                    break;
+                case WebCore::IPAddress::ComparisonResult::CannotCompare:
+                    ASSERT_NOT_REACHED();
+                    return nullptr;
+                }
+            }
+        }
+
+        if (list[upper].contains(address))
+            return &list[upper];
+
+        if (upper != lower && list[lower].contains(address))
+            return &list[lower];
+
+        return nullptr;
+    }
+
+private:
+    static Vector<TrackerAddressLookupInfo>& version4List()
+    {
+        // FIXME: We need to keep this fallback list around not only until the WebPrivacy framework lands
```

Comment:
I don't think this FIXME is still relevant.

Reply:
_Whoops_, definitely no longer relevant! Removed.